### DISCUSSION TITLE
Error in PageSpeed

### DISCUSSION THREAD

**Discussion by VikashPR** [lowlighter/metrics#935] [Q&A] [answered]
Hello, I'am trying to setup pageSpeed insight for the first time. I followed all the given steps in documentation but I'am getting image error that is created by actions.
![Screenshot 2022-03-09 at 8 02 02 AM](https://user-images.githubusercontent.com/69889418/157361197-2c4a7f37-0103-4e38-b4dc-6002ab838f10.png)

This is my repo file : https://github.com/VikashPR/VikashPR/blob/main/github-metrics.svg

**Comment by lowlighter** [comment, marked as answer]
Hi 👋 !

You probably need to setup a [`plugin_pagespeed_token`](https://github.com/lowlighter/metrics/blob/master/source/plugins/pagespeed/README.md#plugin_pagespeed_token), because Google probably applies the rate limit on GitHub's IP addresses.
See https://developers.google.com/speed/docs/insights/v5/get-started to create an api key

Hope it helps 👍

**Reply by VikashPR** [reply]
What do I need to do after creating my API key?

**Reply by lowlighter** [reply]
Just add `plugin_pagespeed_token: ${{ secrets.PAGESPEED_TOKEN }}` to your config and store your api key in your repository secrets under `PAGESPEED_TOKEN` 👍

**Reply by VikashPR** [reply]
And thank you soo much for all the plugins @lowlighter.

**Reply by lowlighter** [reply]
Glad you made it work 🥳 !

Thanks a lot for your support 🙏

**Comment by VikashPR** [comment]
@lowlighter Thank you soo much, it's working now.
I've added `plugin_pagespeed_token: ${{ secrets.PAGESPEED_TOKEN }` in metrics.yml
